src/core/utilities.cpp

```diff
@@ -231,6 +236,23 @@ QString GetTemporaryFileName() {
   return file;
 }
 
+QString SaveToTemporaryFile(const QByteArray& data) {
+  QString filename = GetTemporaryFileName();
```

Comment:
You should just directly use `QTemporaryFile` here to avoid any race conditions.